app/models/maintenance_tasks/run.rb

```diff
@@ -177,6 +179,20 @@ def cancel
     def stuck?
       cancelling? && updated_at <= 5.minutes.ago
     end
+
+    # Performs validation on the presence of a :csv_file attachment.
+    # A Run for a Task that uses CsvCollection must have an attached :csv_file
+    # to be valid. Conversely, a Run for a Task that doesn't use CsvCollection
+    # should not have an attachment to be valid. The appropriate error is added
+    # if the Run does not meet the above criteria.
+    def csv_attachment_presence
+      return if errors[:task_name].any?
```

Comment:
If we've already failed validation because the Task name did not exist, calling `Task.named` will blow up, so we should just return early.

Reply:
I'd rather rescue `Task::NotFoundError` than rely on the other validation having run before.